### ISSUE TITLE
how to insert a full word in the buffer while completing

### ISSUE THREAD

**Issue by toshjp** [company-mode/company-mode#601]
Hello,

I use completion for editing plain documents, not for programming. So when I try to complete a word, I would like a full word rather than a fragment of a word is inserted in the buffer while completing.

When I hit <tab> to start completion, the common part is inserted in the buffer and it is when I hit <return> or type one of the characters that trigger auto-completion that the full word is actually inserted in the buffer.

This is how it works exactly. I type a several characters, hit <tab> several times and then hit <space> to finish completion and continue. When I hit <tab> to start completion, the common part is inserted and then when I hit <space>, both the remaining part and a space is inserted at once.

I'm wondering if it is possible to have a full word inserted in the buffer as I select successive candidates. If completion works in that way, I always see a full word inserted in the buffer while completing and when I hit <space> to continue, nothing but a space is inserted just as I hit the key. This seems to me more like typing documents in a normal way.

I'm afraid this behaviour might annoy those who have got accustomed to the current behaviour. Maybe, some customization could be made possible if that is not too much trouble? Or am I missing something?

Best regards.

**Comment by toshjp** [comment]
Sorry. Some words have been dropped in the last post because they were treated as tags. I post it again.


Hello,

I use completion for editing plain documents, not for programming. So when I try to complete a word, I would like a full word rather than a fragment of a word is inserted in the buffer while completing.

When I hit tab to start completion, the common part is inserted in the buffer and it is when I hit return or type one of the characters that trigger auto-completion that the full word is actually inserted in the buffer.

This is how it works exactly. I type a several characters, hit tab several times and then hit space to finish completion and continue. When I hit tab to start completion, the common part is inserted and then when I hit space, both the remaining part and a space is inserted at once.

I'm wondering if it is possible to have a full word inserted in the buffer as I select successive candidates. If completion works in that way, I always see a full word inserted in the buffer while completing and when I hit space to continue, nothing but a space is inserted just as I hit the key. This seems to me more like typing documents in a normal way.

I'm afraid this behaviour might annoy those who have got accustomed to the current behaviour. Maybe, some customization could be made possible if that is not too much trouble? Or am I missing something?

Best regards.

**Comment by dgutov** [comment]
> I type a several characters, hit tab several times and then hit space to finish completion and continue.

You can try `(setq company-auto-complete t)`. Or follow #526.

**Comment by toshjp** [comment]
> You can try (setq company-auto-complete t). Or follow #526.

Thank you for the quick response.

I had already enabled company-auto-complete. So I looked for other solutions following your pointer. I can see the implementation of the so-called "auto-complete like" interface is not so simple as I imagined, especially because you intend to keep your code clean and tidy.

I had the same question aaronjensen asked as to the reason to avoid modifying the buffer until the end. I thought having each candidate inserted in the buffer as I cycle through all candidates is much simpler and does not tend to divert attention from the point as I type (I also like dabbrev for the same reason), though it is not quite so simple actually.

For now, I will try out some solutions starting with those you described on the page "Switching from AC" (which seem to work fairly well). Thanks for your help.
